hearthstone/player.py

```diff
@@ -112,7 +112,7 @@ def validate_summon_from_hand(self, index: HandIndex, targets: Optional[List[Boa
         if targets is None:
             targets = []
         #  TODO: Jack num_battlecry_targets should only accept 0,1,2
-        if index not in range(len(self.hand)):
+        if not self.valid_hand_index(index):
```

Comment:
I like the `valid_*` naming better than the `validate_*` naming, because it more accurately reflects what the function is doing (answering a boolean question).  You could change all our other validate functions to be named this way  (use the pycharm refactoring tools to do this!).